foam_controlwrapper/foam_files.py

```diff
@@ -80,14 +112,444 @@ def writeFoamFiles(self, caseDirectory, solver):
 
         """
 
-        self.createDirectories(caseDirectory)
-        fileContent = self.createFileContent(solver)
+        self.create_directories(caseDirectory)
+        fileContent = self.create_file_content(caseDirectory, solver,
+                                               time, writeFields)
         for file in fileContent:
             try:
```

Comment:
I do not think that the wrtie code needs to be not enclosed inside a try..except block. The original IOError should have all the necessary infromation for the user to undetify what went wrong, we should not need to create our own IOError.